backend/typescript/middlewares/validators/campValidators.ts

```diff
@@ -0,0 +1,62 @@
+import { Request, Response, NextFunction } from "express";
+import {
+  getApiValidationError,
+  validateArray,
+  validatePrimitive,
+} from "./util";
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable-next-line import/prefer-default-export */
+export const createCampDtoValidator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  if (!validatePrimitive(req.body.name, "string")) {
+    return res.status(400).send(getApiValidationError("name", "string"));
+  }
+  if (
+    req.body.description &&
+    !validatePrimitive(req.body.description, "string")
+  ) {
+    return res.status(400).send(getApiValidationError("description", "string"));
+  }
+  if (!validatePrimitive(req.body.location, "string")) {
+    return res.status(400).send(getApiValidationError("location", "string"));
+  }
+  if (!validatePrimitive(req.body.capacity, "integer")) {
+    return res.status(400).send(getApiValidationError("capacity", "integer"));
+  }
+  if (!validatePrimitive(req.body.fee, "integer")) {
+    return res.status(400).send(getApiValidationError("fee", "integer"));
+  }
+  if (req.body.camperInfo && !validateArray(req.body.camperInfo, "string")) {
+    return res
+      .status(400)
+      .send(getApiValidationError("camperInfo", "string", true));
+  }
+  if (req.body.campers && !validateArray(req.body.campers, "string")) {
+    return res
+      .status(400)
+      .send(getApiValidationError("campers", "string", true));
+  }
+  if (req.body.waitlist && !validateArray(req.body.waitlist, "string")) {
+    return res
+      .status(400)
+      .send(getApiValidationError("waitlist", "string", true));
+  }
+  if (!validatePrimitive(req.body.startDate, "string")) {
+    return res.status(400).send(getApiValidationError("startDate", "string"));
+  }
+  if (!validatePrimitive(req.body.endDate, "string")) {
+    return res.status(400).send(getApiValidationError("endDate", "string"));
+  }
```

Comment:
can we validate that these are date strings? Udy added a validateDate in her PR, we could use this once her changes are merged in https://github.com/uwblueprint/focus-on-nature/pull/14/files#diff-be83b7a804417800c6eb80c3b0efcc7b9bf3982372032ee49951bcb2d485e4d5R59

Reply:
oh cool ! added another validator for times of day (i.e. 12:59)